nucypher/blockchain/eth/actors.py

```diff
@@ -1045,23 +1037,26 @@ def generate_policy_parameters(self,
             raise ValueError("Policy end time must be specified as 'expiration' or 'duration_periods', got neither.")
 
         # Merge injected and default params.
-        first_period_reward = first_period_reward or self.first_period_reward
         rate = rate or self.rate
         duration_periods = duration_periods or self.duration_periods
 
         # Calculate duration in periods and expiration datetime
-        if expiration:
-            duration_periods = calculate_period_duration(future_time=expiration,
-                                                         seconds_per_period=self.economics.seconds_per_period)
-        else:
-            duration_periods = duration_periods or self.duration_periods
+        if duration_periods:
+            # Duration equals one period means that expiration date is the last second of the current period
             expiration = datetime_at_period(self.staking_agent.get_current_period() + duration_periods,
-                                            seconds_per_period=self.economics.seconds_per_period)
+                                            seconds_per_period=self.economics.seconds_per_period,
+                                            start_of_period=True)
+            expiration -= 1  # Get the last second of the target period
+        else:
+            now = self.staking_agent.blockchain.w3.eth.getBlock(block_identifier='latest').timestamp
+            duration_periods = calculate_period_duration(now=maya.MayaDT(now),
+                                                         future_time=expiration,
+                                                         seconds_per_period=self.economics.seconds_per_period)
+            duration_periods += 1  # Number of all included periods
```

Comment:
Are we using `+=1` because the current day counts as one? Should `calculate_period_duration` make this adjustment, instead of having to manually do it?

Reply:
Yeah, you're right, `calculatePeriodDuration` returns `periodAt(expiration) - periodAt(now)` and we add current period.
> Should `calculate_period_duration` make this adjustment, instead of having to manually do it?

Good question, I'm still not sure what behaviour is more straightforward

Reply:
It would seem that it should to me, but it may depend on where else it is used, and whether that same `+=1` logic applies.

Reply:
`calculatePeriodDuratio` used in `initialize_stake` without adding 1